STL_Adapter/CRUD/Create.cs

```diff
@@ -0,0 +1,124 @@
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+
+using BH.oM.Adapter;
+
+using BH.oM.STL;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+
+namespace BH.Adapter.STL
+{
+    public partial class STLAdapter : BHoMAdapter
+    {
+        protected override bool ICreate<T>(IEnumerable<T> objects, ActionConfig actionConfig = null)
+        {
+            return Create(objects as dynamic);
+        }
+
+        private bool Create(IEnumerable<GeometryGroup> geometryGroups)
+        {
+            foreach(GeometryGroup g in geometryGroups)
+            {
+                string stlName = g.FileName;
+                List<string> geoTxt = new List<string>();
+                List<Polyline> polylines = g.Geometry.SelectMany(x => x.ToPolyline()).ToList();
+
+                if (polylines.Any())
+                {
+                    geoTxt.Add("solid " + stlName + ".stl");
+                    foreach (Polyline polyline in polylines)
+                    {
+                        try
+                        {
+                            geoTxt.AddRange(polyline.ToSTL(m_stlSettings));
+                        }
+                        catch (Exception e)
+                        {
+                            BH.Engine.Reflection.Compute.RecordError("An error occurred in exporting the STL file. Error is: " + e.ToString());
+                            return false;
+                        }
+                    }
+                    geoTxt.Add("endsolid " + stlName.ToString() + ".stl");
+                }
+
+                try
+                {
+                    // write text files
+                    File.WriteAllLines(System.IO.Path.Combine(m_stlSettings.Directory, stlName, ".stl"), geoTxt);
+                }
+                catch (Exception e)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("An error occurred in exporting the STL file. Error is: " + e.ToString());
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private bool Create(IEnumerable<Polyline> polylines, string fileName = null)
```

Comment:
Do we need this Create method for polylines? A polyline cannot be converted to a valid STL file, unless it is a closed polyline that describes a surface, in which case we should use GeometryGroup.

Reply:
Yes this is for other representation workflows outside of building environments

Reply:
Got it. I will leave it to you to decide but just would like to suggest, maybe we should add a IsClosed check to the polyline? Because by definition of STL file, a open polyline is not valid, although it may still write the file.